### PR TITLE
Check data before merging 

### PR DESCRIPTION
**Pull request by SudoGetBeer**
As a extension suggested by @bastien-phi here https://github.com/hotmeteor/spectator/pull/89#issuecomment-1014272585

Now it check the basic types if they are the correct type.

### CODE REVIEW

File: src/Validation/RequestValidator.php [hunk 1]
```diff
@@ -122,7 +122,22 @@ protected function validateParameters()
...
+                        switch ($expected_parameter_schema->type){
+                            case 'integer':
+                            case 'number':
+                                $typeIsCorrect = is_numeric($actual_parameter);
+                                break;
+                            case 'string':
+                                $typeIsCorrect = is_string($actual_parameter);
+                                break;
+                            case 'boolean':
+                                $typeIsCorrect = filter_var($actual_parameter, FILTER_VALIDATE_BOOLEAN) !== false;
```
Comment (bastien-phi) [inline src/Validation/RequestValidator.php:135]: in openapi, booleans are `'true'` and `'false'` only
(https://swagger.io/docs/specification/data-models/data-types/#boolean, https://spec.openapis.org/oas/v3.1.0#data-types)
Using `filter_var` here can lead to false positive as `'on'` will be considered as valid boolean (which is not)

File: src/Validation/RequestValidator.php [hunk 2]
```diff
@@ -122,7 +122,22 @@ protected function validateParameters()
...
+                                break;
+                        }
+                        if ($typeIsCorrect){
+                            settype($actual_parameter, $expected_parameter_schema->type);
```
Comment (bastien-phi) [inline src/Validation/RequestValidator.php:139]: settype won't work for booleans : 

```php
$actual_parameter = 'false';
settype($actual_parameter, 'boolean');
var_dump($actual_parameter); // bool(true)
```

**Comment by bastien-phi** [comment]
See https://github.com/SudoGetBeer/spectator/pull/1 for the patch

**Comment by SudoGetBeer** [comment]
@bastien-phi thanks! Merged it.

**Comment by hotmeteor** [comment]
@SudoGetBeer Can you sort out the StyleCI stuff on this one?

**Pull request merged**